src/com/trilead/ssh2/crypto/dh/DhGroupExchange.java

```diff
@@ -87,10 +86,10 @@ public void setF(BigInteger f)
 		this.k = f.modPow(x, p);
 	}
 
-	public byte[] calculateH(byte[] clientversion, byte[] serverversion, byte[] clientKexPayload,
-			byte[] serverKexPayload, byte[] hostKey, DHGexParameters para)
```

Comment:
For API compatibility, keep the original method and just make it delegate to your new method with a hard-coded "SHA1" `hashAlgo`